internal/server/RoundLoop.go

```diff
@@ -65,19 +65,52 @@ func (s *Server) RunBikeSwitch(gameState GameStateDump) {
 
 func (s *Server) HandleKickoutProcess() {
 	for _, bike := range s.GetMegaBikes() {
-		agentsVoteCounts := bike.KickOutAgent()
-		for agentID, votes := range agentsVoteCounts {
-			if votes > len(bike.GetAgents())/2 {
-				bike.RemoveAgent(agentID)
-				delete(s.megaBikeRiders, agentID)
-
-				if agent, ok := s.GetAgentMap()[agentID]; ok {
-					agent.ToggleOnBike()
-					//if kickedoutagent is the leader, need to select one new
-				}
+		agentsVotes := make([]uuid.UUID, 0)
+
+		// the kickout process only happens democratically in level 0 and level 1
+		switch bike.GetGovernance() {
+		case utils.Democracy:
+			// make map of weights of 1 for all agents on bike
+			agents := bike.GetAgents()
+			weights := make(map[uuid.UUID]float64)
+			for _, agent := range agents {
+				weights[agent.GetID()] = 1.0
 			}
 
+			// get which agents are getting kicked out
+			agentsVotes = bike.KickOutAgent(weights)
+
+		case utils.Leadership:
+			// get the map of weights from the leader
+			leader := s.GetAgentMap()[bike.GetRuler()]
+			weights := leader.DecideWeights(utils.Kickout)
+			// get which agents are getting kicked out
+			agentsVotes = bike.KickOutAgent(weights)
+
+		case utils.Dictatorship:
+			// in level 2 only the ruler can kick out people
+			dictator := s.GetAgentMap()[bike.GetRuler()]
+			agentsVotes = dictator.DecideKickOut()
 		}
+
+		// perform kickout
+		leaderKickedOut := false
+		for _, agentID := range agentsVotes {
+			bike.RemoveAgent(agentID)
+			delete(s.megaBikeRiders, agentID)
+			if agent, ok := s.GetAgentMap()[agentID]; ok {
+				agent.ToggleOnBike()
+			}
```

Comment:
We do this in a few places now - it should probably be pulled out to a function, but don't worry about it for this PR